embedded-graphics/src/geometry/angle.rs

```diff
@@ -0,0 +1,477 @@
+use super::real;
+use crate::geometry::Real;
+use core::f32::consts::PI;
+use core::ops::{Add, AddAssign, Neg, Sub, SubAssign};
+use float_cmp::{ApproxEq, F32Margin};
+#[cfg(not(feature = "fixed_point"))]
+#[allow(unused_imports)]
+use micromath::F32Ext;
+
+pub(crate) mod angle_consts {
+    use super::{real, Angle};
+
+    pub(crate) const ANGLE_90DEG: Angle = Angle(real::FRAC_PI_2);
+    pub(crate) const ANGLE_180DEG: Angle = Angle(real::PI);
+    pub(crate) const ANGLE_360DEG: Angle = Angle(real::TAU);
+}
+
+/// Angle.
+///
+/// `Angle` is used to define the value of an angle.
+///
+/// # Examples
+///
+/// ## Create an `Angle` from a value
+///
+/// ```rust
+/// use embedded_graphics::geometry::Angle;
+///
+/// // Create an angle using the `from_degree` constructor method
+/// let angle_a = Angle::from_degree(10.0);
+/// ```
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct Angle(Real);
+
+impl Angle {
+    /// Creates an angle defined in degrees.
+    pub fn from_degree(angle: f32) -> Self {
+        Angle((angle * PI / 180.0).into())
+    }
+
+    /// Creates an angle defined in radian.
+    pub fn from_radian(angle: f32) -> Self {
+        Angle(angle.into())
+    }
+
+    /// Creates a zero degree angle.
+    pub fn zero() -> Self {
+        Angle(0.into())
+    }
+
+    /// Compute the absolute value of the angle.
+    pub fn abs(self) -> Self {
+        Angle(self.0.abs())
+    }
+
+    /// Normalize the angle to less than one full rotation (ie. in the range 0..360).
+    pub fn normalize(self) -> Self {
+        Angle(self.0.rem_euclid((2.0 * PI).into()))
+    }
+
+    /// Normalize the angle to less than one full rotation, starting from angle.
+    /// (ie. in the range angle..(angle+360)).
+    pub(crate) fn normalize_from(self, angle: Self) -> Self {
+        Angle((self.0 - angle.0).rem_euclid((2.0 * PI).into()) + angle.0)
+    }
+
+    /// Return numerical value of the angle in degree
+    pub fn to_degree(self) -> f32 {
+        let angle: f32 = self.0.into();
+        180.0 * angle / PI
+    }
+
+    /// Return numerical value of the angle in radian
+    pub fn to_radian(self) -> f32 {
+        self.0.into()
+    }
+}
+
+/// AngleUnit trait.
+///
+/// `AngleUnit` is a trait to convert numbers into angle by appending .deg() or .rad()
+/// to the number as if it was a unit.
+///
+/// # Examples
+///
+/// ## Create an `Angle` from a value using `AngleUnit`
+///
+/// ```rust
+/// use embedded_graphics::geometry::AngleUnit;
+///
+/// // Create an angle using the `AngleUnit` method
+/// let angle_b = 30.0.deg();
```

Comment:
For completeness' sake, it would be great if this example showed usage of `.rad()` as well.